src/linux/init/init.cpp

```diff
@@ -2661,10 +2652,56 @@ try
     }
 
     //
-    // Respond to the instance termination request.
+    // Attempt to stop the plan9 server, if it is not able to be stopped because of an
+    // in-use file, reply to the service that the instance could not be terminated.
     //
 
-    Channel.SendResultMessage<bool>(StopPlan9Server(Message->Force, Config));
+    if (!StopPlan9Server(Message->Force, Config))
+    {
+        Channel.SendResultMessage<bool>(false);
+        return;
+    }
+
+    InitTerminateInstanceInternal(Config);
+}
+CATCH_LOG();
+
+void InitTerminateInstanceInternal(wsl::linux::WslDistributionConfig& Config)
+
+/*++
+
+Routine Description:
+
+    This routine attempts to cleanly terminate the instance.
+
+Arguments:
+
+    Config - Supplies the distribution config.
+
+Return Value:
+
+    None.
+
+--*/
+try
+{
+    if (Config.BootInit && !Config.BootStartWriteSocket)
+    {
+        THROW_LAST_ERROR_IF(UtilSetSignalHandlers(g_SavedSignalActions, false) < 0);
+
+        UtilExecCommandLine("systemctl poweroff", nullptr);
```

Comment:
Looks like this changed from `poweroff` to `reboot` but I guess they both do the same thing ?

Reply:
Also we might want to immediately shutdown if `systemctl poweroff` fails